client/java/src/main/java/io/openlineage/client/utils/jdbc/TeradataJdbcExtractor.java

```diff
@@ -0,0 +1,61 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.utils.jdbc;
+
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+
+public class TeradataJdbcExtractor implements JdbcExtractor {
+  // https://teradata-docs.s3.amazonaws.com/doc/connectivity/jdbc/reference/current/jdbcug_chapter_2.html#URL_DATABASE
+
+  private static final String SCHEME = "teradata";
```

Comment:
By the way, according to the standard: https://www.rfc-editor.org/rfc/rfc3986#section-3.1:
```
 Although schemes are case-
   insensitive, the canonical form is lowercase and documents that
   specify schemes must do so with lowercase letters.  An implementation
   should accept uppercase letters as equivalent to lowercase in scheme
   names
```
which means one should do `jdbcUri.toLowerCase().startsWith(SCHEME.toLowerCase());`

Reply:
This raises a question - should URLs be normalized (lowercase scheme, lowercase authority) before constructing dataset namespace? Currently passing URLs in different cases leads to producing different datasets.

Also? for most RDBMS object names are case insensitive, so users may pass both `SCHEMA.TABLE` and `schema.table` with the same result. Should these be normalized as well?

Reply:
Moved this discussion to https://github.com/OpenLineage/OpenLineage/issues/2829.

Regarding behavior of `isDefined` method, I'd prefer to implement this in a separated PR, because currently URLs like `JDBC:TERADATA://...` are not properly handled by `JdbcDatasetUtils` - it trims prefix `jdbc:`, but not `JDBC:`